core/src/main/resources/AppConstants.properties

```diff
@@ -106,15 +106,44 @@ transactionmanager.narayana.stuckTimerTime=30000
 jdbc.required=true
 jdbc.datasource.default=jdbc/${instance.name.lc}
 
-# Connection Pool properties for BTM and Narayana. Narayana only supports maxPoolSize
+
+## JDBC Connection Pool properties for BTM
+# Minimum number of physical connections to maintain.
+transactionmanager.btm.jdbc.connection.minPoolSize=0
+# Maximum number of physical connections that you can create in this pool.
+transactionmanager.btm.jdbc.connection.maxPoolSize=20
+# Amount of time a connection can be unused or idle until it can be discarded.
+transactionmanager.btm.jdbc.connection.maxIdleTime=60
+# The maximum lifetime that a connection can stay in the pool before getting closed.
+transactionmanager.btm.jdbc.connection.maxLifeTime=0
+
+
+
+## JMS Connection Pool properties for BTM
 # Minimum number of physical connections to maintain.
-jdbc.connection.minPoolSize=0
+transactionmanager.btm.jms.connection.minPoolSize=0
 # Maximum number of physical connections that you can create in this pool.
-jdbc.connection.maxPoolSize=20
-# Amount of time a connection can be unused or idle until it can be discarded
-jdbc.connection.maxIdleTime=60
+transactionmanager.btm.jms.connection.maxPoolSize=20
+# Amount of time a pooled connection can be unused or idle until it can be discarded.
+transactionmanager.btm.jms.connection.maxIdleTime=60
 # The maximum lifetime that a connection can stay in the pool before getting closed.
-jdbc.connection.maxLifeTime=0
+transactionmanager.btm.jms.connection.maxLifeTime=0
+
+
+
+## JDBC Connection Pool properties for Narayana
+# Maximum number of physical connections that you can create in this pool.
+transactionmanager.narayana.jdbc.connection.maxPoolSize=20
+
+## JMS Connection Pool properties for Narayana
+# Maximum number of physical connections that you can create in this pool.
+transactionmanager.narayana.jms.connection.maxPoolSize=20
+# Amount of time a connection can be unused or idle until it can be discarded.
+transactionmanager.narayana.jms.connection.maxIdleTime=60
+# Maximum number of jms sessions per connection that can be created in the connection pool.
+transactionmanager.narayana.jms.session.maxPoolSize=20
```

Comment:
"max nr of sessions per connection" ≠ "session pool size"; there is no pool of sessions. I think it is better to change this property name, for instance `transactionmanager.narayana.jms.connection.maxSessions=500`